File: .tmp_py4013/batch_04_sources/2485_find_the_pivot_integer.js
// LeetCode 2485 - Find the Pivot Integer
// https://leetcode.com/problems/find-the-pivot-integer/

/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function(n) {
    const total = n * (n + 1) / 2;
    let sum = 0;
    for (let x = 1; x <= n; x++) {
        sum += x;
        if (sum === total - sum + x) return x;
    }
    return -1;
};

File: .tmp_rs4013/rust_check_batch08/2485_find_the_pivot_integer.rs
struct Solution;
// LeetCode 2485 - Find the Pivot Integer
// https://leetcode.com/problems/find-the-pivot-integer/

impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let total = n * (n + 1) / 2;
        let mut sum = 0;
        for x in 1..=n {
            sum += x;
            if sum == total - sum + x {
                return x;
            }
        }
        -1
    }
}

fn main() {}
